### ISSUE TITLE
tess4j 4.4.0 crashes on mac osx

### ISSUE THREAD

**Issue by jiangyurong609** [nguyenq/tess4j#163]
I got the following error when I try to run simple ocr (https://github.com/nguyenq/tess4j/wiki/Code-Sample) for an image. Any help would be much appreciated!
```

# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x000000012ef5026e, pid=4056, tid=0x0000000000001a03
#
# JRE version: Java(TM) SE Runtime Environment (8.0_172-b11) (build 1.8.0_172-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.172-b11 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# !strcmp(locale, "C"):Error:Assert failed:in file baseapi.cpp, line 209
C  [libtesseract.dylib+0x15726e]  _ZNK7ERRCODE5errorEPKc16TessErrorLogCodeS1_z+0x178
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /xx/hs_err_pid4056.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.

```

**Comment by nguyenq** [comment]
Duplicate of https://github.com/nguyenq/tess4j/issues/105.

**Comment by jiangyurong609** [comment]
hey @nguyenq thanks, I saw you fixed issue with this commit for 4.1.0, however, the problem still persists in 4.4.0, any suggestion?  https://github.com/tesseract-ocr/tesseract/commit/96f6fc28b80768a220e973f31100c9af725020cf

**Comment by nguyenq** [comment]
It seems you're still using Tesseract 4.0.0. You need to upgrade it to 4.1.0 to avoid this issue.

**Comment by jiangyurong609** [comment]
@nguyenq I'm using 4.4.0. 

![image](https://user-images.githubusercontent.com/3674274/65196830-af427200-da35-11e9-8795-6d20ab48751e.png)

**Comment by nguyenq** [comment]
The locale issue existed in Tesseract 4.0.0; you'll need to upgrade it to Tesseract 4.1.0 to get the fix. It's not a problem with Tess4J.

**Comment by jiangyurong609** [comment]
@nguyenq thank you for the kind help. I upgraded to 4.1.0, but it only works for default engine. 

I got similar error, when I set engine as follows ` instance.setOcrEngineMode(TessAPI.TessOcrEngineMode.OEM_TESSERACT_LSTM_COMBINED);` ```Failed loading language 'eng'
Tesseract couldn't load any languages!
Warning: Invalid resolution 0 dpi. Using 70 instead.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000124afdd75, pid=14756, tid=0x0000000000002703
#
# JRE version: Java(TM) SE Runtime Environment (8.0_172-b11) (build 1.8.0_172-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.172-b11 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libtesseract.dylib+0x17d75]  tesseract::Tesseract::recog_all_words(PAGE_RES*, ETEXT_DESC*, TBOX const*, char const*, int)+0xaf
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/yujiang/Desktop/tensorflow/models/samples/languages/java/label_image/hs_err_pid14756.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

**Comment by nguyenq** [comment]
Only Tesseract Windows binaries are bundled with Tess4J; for other platforms, the users would have to install or build Tesseract themselves. That's stated in the documentation.

Use `tessdata_fast` or `tessdata_best` for LSTM engine mode.

https://github.com/tesseract-ocr

**Comment by balmma** [comment]
> I got similar error, when I set engine as follows instance.setOcrEngineMode(TessAPI.TessOcrEngineMode.OEM_TESSERACT_LSTM_COMBINED);

This is because the bundled eng.traineddata in Tess4J is from https://github.com/tesseract-ocr/tessdata_best and doesn't include the legacy models. If you want to use OEM_TESSERACT_ONLY or OEM_TESSERACT_LSTM_COMBINED you have to use *.traineddata from https://github.com/tesseract-ocr/tessdata.

**Comment by balmma** [comment]
@nguyenq 

Just another question:

What's the reason to include the pure tessdata_best (non integerized) traineddata in Tess4J?

Wouldn't it be more appropriate to include either an integerized version of tessdata_best or even tessdata_fast? Perfomance would be 4 to 5 times better without any serious compromise in accuracy. Additionally the traineddata model size is much smaller.

**Comment by balmma** [comment]
We have an ongoing discussion about this topic in the SikuliX project:
https://github.com/RaiMan/SikuliX1/commit/10eb3798961cd5f13b85bf209ad52efe9d7ff3ab

**Comment by balmma** [comment]
Have just opened a new issue for this. Sorry for the SPAM :-)
